### PR TITLE
Api-Gateway authentication control for paths

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Extend Api-Gateway API with attributes that allow to control JWT validation based on request path. See Istio [docs](https://istio.io/docs/tasks/security/authn-policy/#disable-end-user-authentication-for-specific-paths).

Changes proposed in this pull request:

- Update Api-Gateway documentation

**Related issue(s)**
See also #4262

### CODE REVIEW

File: docs/api-gateway/03-01-security.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ When you expose a service in Kyma, you can secure it by specifying the **authent
 
 If you don't specify any of these attributes, the API Controller doesn't create an Istio Authentication Policy for the service and leaves it unsecured.
 
->**NOTE:** You can secure only the entire service. You cannot secure the specific endpoints of the service.
+>**NOTE:** You can disable JWT validation for specific resource paths. This makes such resources unsecured. see [this](#custom-resource-api) document for details.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/api-gateway/03-01-security.md]: wrong link

File: docs/api-gateway/06-01-api.md [hunk 2]
```diff
@@ -53,6 +60,32 @@ These are the possible scenarios depending on the values the the **spec.authenti
 - If you set the value to `true` and the **spec.authentication** parameter value is not empty, authentication is enabled for the service. Entries from the **spec.authentication** parameter are used to create the value of the **origins** attribute of the corresponding Istio Authentication Policy.
 - If you set the value to `true` and the **spec.authentication** parameter value is empty or not provided, authentication is enabled for the service. The default authentication settings specified in API Controller environment variables are used to create the value of the **origins** attribute of the corresponding Istio Authentication Policy.
 
+### Configuration of resources excluded from JWT validation
```
Comment (tomekpapiernik) [inline docs/api-gateway/06-01-api.md]: Please remove this section, I added it to `03-01` instead.

File: docs/api-gateway/06-01-api.md [hunk 3]
```diff
@@ -42,6 +47,8 @@ This table lists all the possible parameters of a given resource together with t
 | **spec.authentication** | **NO** | Allows to specify an array of authentication policies that secure the service. |
 | **authentication.type** | **YES** | Specifies the desired authentication method that secures the exposed service. |
 | **authentication.jwt.issuer**, **authentication.jwt.jwksUri** | **YES** | Specifies the issuer of the tokens used to access the services, as well as the JWKS endpoint URI. |
+| **authentication.jwt.triggerRule** | **NO** | Specifies rules for disabling JWT validation for specific resource paths. If this parameter value is empty or not provided, JWT validation is always enabled. |
```
Comment (tomekpapiernik) [inline docs/api-gateway/06-01-api.md]:
```suggestion
| **authentication.jwt.triggerRule** | **NO** | Specifies rules for disabling JWT validation for specific resource paths. If this parameter value is empty or not provided, JWT validation is enabled for all resource paths. |
```

File: docs/api-gateway/06-01-api.md (hunk 2 above)
Comment (tomekpapiernik) [inline docs/api-gateway/06-01-api.md]: Please remove this section, I added it to `03-01` instead.

File: docs/api-gateway/06-01-api.md [hunk 4]
```diff
@@ -42,6 +47,8 @@ This table lists all the possible parameters of a given resource together with t
...
+| **authentication.jwt.triggerRule.excludedPaths** | **NO** | Specifies a list of matching expressions that configure resource paths excluded from JWT validation.  See [this](#configuration-of-resources-excluded-from-JWT-validation) section for more details. |
```
Comment (tomekpapiernik) [inline docs/api-gateway/06-01-api.md]:
```suggestion
| **authentication.jwt.triggerRule.excludedPaths** | **NO** | Specifies a list of matching expressions that configure resource paths excluded from JWT validation.  See [this](#details-security-specify-service-resource-paths-not-sercured-with-jwt-authentication) document for more details. |
```

File: docs/api-gateway/06-01-api.md (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply docs/api-gateway/06-01-api.md]: Done.

**Pull request merged**